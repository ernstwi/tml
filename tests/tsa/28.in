default location
    invar { x < 5 }

default edge
    reset { x }

init foo
foo -> bar

default location
    invar { y < 10 }

default edge
    reset { y }

baz
bar -> baz
