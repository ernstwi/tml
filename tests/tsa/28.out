{
    "actions": [],
    "clocks": [
        "x",
        "y"
    ],
    "edges": [
        {
            "action": null,
            "from": "bar",
            "guard": null,
            "reset": [
                "y"
            ],
            "to": "baz"
        },
        {
            "action": null,
            "from": "foo",
            "guard": null,
            "reset": [
                "x"
            ],
            "to": "bar"
        }
    ],
    "locations": [
        {
            "id": "bar",
            "initial": false,
            "invariant": [
                {
                    "clock": "x",
                    "cmp": "lt",
                    "n": 5
                }
            ]
        },
        {
            "id": "baz",
            "initial": false,
            "invariant": [
                {
                    "clock": "y",
                    "cmp": "lt",
                    "n": 10
                }
            ]
        },
        {
            "id": "foo",
            "initial": true,
            "invariant": [
                {
                    "clock": "x",
                    "cmp": "lt",
                    "n": 5
                }
            ]
        }
    ]
}
